### ISSUE TITLE
RFE: please provide support for llvm 16

### ISSUE THREAD

**Issue by kloczek** [iovisor/bpftrace#2576]
Lookd like currnyly llvm 16.x is no tsupported.
```console
CMake Error at CMakeLists.txt:171 (message):
  Unsupported LLVM version found via /usr/include: 16


CMake Error at CMakeLists.txt:172 (message):
  Only versions between 6 and 15 are supported


CMake Error at CMakeLists.txt:173 (message):
  Specify an LLVM major version using LLVM_REQUESTED_VERSION=<major version>
```

**Comment by viktormalik** [comment]
It's already being solved in #2528 but there's a bug in LLVM 16 that prevents building bpftrace with it.

**Comment by kloczek** [comment]
OK I'm closing this ticket as looks like everything is already in WIP state
I've subscribed to PR and LLVM ticket.

Thank you 👍
